src/com/sap/piper/Utils.groovy

```diff
@@ -113,3 +113,13 @@ static String fillTemplate(String templateText, Map binding){
     String result = engine.createTemplate(templateText).make(binding)
     return result
 }
+
+static String downloadMavenSettingsFromUrlIfRequired(script, String settingsFileLocator) {
+    if (settingsFileLocator.startsWith("http")) {
+        def settings = script.httpRequest settingsFileLocator
+        def settingsFile = 'settings.xml'
```

Comment:
last invocation wins (in terms of intended side effects) when invoked multiple times with remote files